src/libstd/os/linux/raw.rs

```diff
@@ -205,7 +205,8 @@ mod arch {
     }
 }
 
-#[cfg(target_arch = "x86_64")]
+#[cfg(any(target_arch = "x86_64", target_arch = "powerpc64",
+          target_arch = "powerpc64le"))]
```

Comment:
Have you checked that the sizes of all these types are correct?  (They probably are, but it's worth double-checking.)